feat: support multiple project setup projects

Jefiozie opened the pull request:
Will update the description after our call

bampakoa commented on schematics/utils.ts:
```diff
@@ -0,0 +1,31 @@
+import { Tree } from '@angular-devkit/schematics';
+
+// this code is orginal from @schematics/angular
+// we also could include the lib and provide the func getWorkspace from their
```
I would definitely recommend to use it directly from `@schematics/angular` 💯

Jefiozie replied:
Let's discuss this in our session 😀

bampakoa commented:
@Jefiozie this is what I had in mind from your contribution at the first place 🥰 . You see, I am left a bit behind with schematics and I used some old techniques for getting the workspace, project and so on. But I can see from your changes that the API has really changed and became much simpler. Thanks so much for this 🙏

bampakoa commented on package.json:
```diff
@@ -12,6 +12,9 @@
   "main": "dist/index.js",
   "builders": "builders.json",
+  "ng-add": {
+    "save": "dependencies"
```
Does this mean that essentially that `ng add` will also install it as a dependency?

Jefiozie replied:
Yes it does, let me show you the part in the `angular-cli` that does this.

bampakoa commented:
@Jefiozie could you please rebase that too? 🙏

Jefiozie commented:
I've rebased it and I've also added the `getWorkspace` function again. In the `@schematics/angular` package there is no host returned, this is needed to update the `angular.json` file. Maybe we can refactor this in a later stadium?

bampakoa commented on schematics/index.ts:
```diff
@@ -1,45 +1,41 @@
   Tree
 } from '@angular-devkit/schematics';
-import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { Schema } from "./schema";
```
```suggestion
import { Schema } from './schema';
```

bampakoa commented on schematics/index.ts:
```diff
@@ -1,45 +1,41 @@
-    throw new SchematicsException();
+  if (project.extensions.projectType !== 'application') {
+    throw new SchematicsException(
+      `Electron requires an Angular project type of "application" in angular.json.`
```
```suggestion
      `ngx-electronify requires an Angular project type of "application" in angular.json.`
```

bampakoa commented on schematics/utility.ts:
```diff
@@ -0,0 +1,34 @@
+import { virtualFs, workspaces } from "@angular-devkit/core";
+import { Tree } from "@angular-devkit/schematics";
```
```suggestion
import { virtualFs, workspaces } from '@angular-devkit/core';
import { Tree } from '@angular-devkit/schematics';
```

bampakoa reviewed (changes requested):
Sure @Jefiozie 👍 I have added some suggestions and we are good to go 🚀

Jefiozie commented:
We're all set now!

bampakoa reviewed (approved):
@Jefiozie I took the initiative and merged master to the PR because `package-lock.json` had conflicts again.

Jefiozie commented:
Thank you, let me know if I need to change something
